src/Sarif.UnitTests/Visitors/InsertOptionalDataVisitorTests.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Microsoft.CodeAnalysis.Sarif.UnitTests;
+using Newtonsoft.Json;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Sarif.Visitors
+{
+    public class InsertOptionalDataVisitorTests : FileDiffingTests
+    {
+        // To rebaseline all test files set this value to true and rerun the testa
+        private static bool s_rebaseline = false;
+
+        private ITestOutputHelper _outputHelper;
+
+        private static string GetTestDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@".\TestData", subdirectory));
+        }
+
+        private static string GetProductTestDataDirectory(string subdirectory = "")
+        {
+            return Path.GetFullPath(Path.Combine(@"..\..\..\..\..\\src\Sarif.UnitTests\TestData", subdirectory));
+        }
+
+        public InsertOptionalDataVisitorTests(ITestOutputHelper outputHelper)
+        {
+            _outputHelper = outputHelper;
+        }
+
+        [Theory]
+        [InlineData(OptionallyEmittedData.Hashes)]
+        [InlineData(OptionallyEmittedData.TextFiles)]
+        [InlineData(OptionallyEmittedData.RegionSnippets)]
+        [InlineData(OptionallyEmittedData.ComprehensiveRegionProperties)]
+        [InlineData(OptionallyEmittedData.ComprehensiveRegionProperties | OptionallyEmittedData.RegionSnippets | OptionallyEmittedData.TextFiles | OptionallyEmittedData.Hashes)]
+        public void InsertOptionaDataVisitorTests_InsertsOptionalDataForCommonConditions(OptionallyEmittedData optionallyEmittedData)
```

Comment:
>InsertOptionaDataVisitorTests [](start = 20, length = 29)

Nit: To make the test name read in English ("The InsertOptionalDataVistor inserts optional data for common conditions"), the test name would be `InsertOptionaDataVisitor_InsertsOptionalDataForCommonConditions`.

Reply:
Sure, just a typo 'OptionaData' is not a real thing :)

---
In reply to: [205982390](https://github.com/Microsoft/sarif-sdk/pull/974#discussion_r205982390) [](ancestors = 205982390)